projects/schematics/src/migrations/4_0/constructor-deprecations/data/configurator-storefront-utils.service.migration.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  ANGULAR_CORE,
+  CONFIGURATOR_GROUPS_SERVICE,
+  CONFIGURATOR_STOREFRONT_UTILS_SERVICE,
+  KEYBOARD_FOCUS_SERVICE,
+  PLATFORM_ID_STRING,
+  SPARTACUS_PRODUCT_CONFIGURATOR_RULEBASED,
+  SPARTACUS_STOREFRONTLIB,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const CONFIGURATOR_STOREFRONT_UTILS_SERVICE_MIGRATION: ConstructorDeprecation = {
+  // feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts
+  class: CONFIGURATOR_STOREFRONT_UTILS_SERVICE,
+  importPath: SPARTACUS_PRODUCT_CONFIGURATOR_RULEBASED,
+  deprecatedParams: [
+    {
+      className: CONFIGURATOR_GROUPS_SERVICE,
+      importPath: SPARTACUS_PRODUCT_CONFIGURATOR_RULEBASED,
+    },
+    {
+      className: PLATFORM_ID_STRING,
+      importPath: ANGULAR_CORE,
+    },
```

Comment:
```suggestion
   {
      className: PLATFORM,
      literalInference: ANY_TYPE,
      injectionToken: {
        token: PLATFORM_ID_STRING,
        importPath: ANGULAR_CORE,
      },
    }
```
(might need to run prettier)

Reply:
No, this does not work. I also found this technique but saw that the constructor-deprecations mechanism in this case does not find the constructors it needs to update. Please see branch https://github.com/SAP/spartacus/tree/feature/GH-12198-testsForConstructorDeprecations, and run `yarn test `in the schematics folder